src/fread.c

```diff
@@ -34,22 +34,33 @@ static char quote, dec;
 static int linesForDecDot; // when dec='auto', track the balance of fields in favor of dec='.' vs dec=',', ties go to '.'
 static bool eol_one_r;  // only true very rarely for \r-only files
 
-// Quote rule:
-//   0 = Fields may be quoted, any quote inside the field is doubled. This is
-//       the CSV standard. For example: <<...,"hello ""world""",...>>
-//   1 = Fields may be quoted, any quotes inside are escaped with a backslash.
-//       For example: <<...,"hello \"world\"",...>>
-//   2 = Fields may be quoted, but any quotes inside will appear verbatim and
-//       not escaped in any way. It is not always possible to parse the file
-//       unambiguously, but we give it a try anyways. A quote will be presumed
-//       to mark the end of the field iff it is followed by the field separator.
-//       Under this rule eol characters cannot appear inside the field.
-//       For example: <<...,"hello "world"",...>>
-//   3 = Fields are not quoted at all. Any quote characters appearing anywhere
-//       inside the field will be treated as any other regular characters.
-//       Example: <<...,hello "world",...>>
-//
-static int quoteRule;
+enum quoteRule
+{
+    // Fields may be quoted, any quote inside the field is doubled.This is
+    // the CSV standard. For example: <<...,"hello ""world""",...>>
+    QUOTE_RULE_EMBEDDED_QUOTES_DOUBLED,
+
+    // Fields may be quoted, any quotes inside are escaped with a backslash.
+    // For example: <<...,"hello \"world\"",...>>
+    QUOTE_RULE_EMBEDDED_QUOTES_ESCAPED,
+
+    // Fields may be quoted, but any quotes inside will appear verbatim and
+    // not escaped in any way. It is not always possible to parse the file
+    // unambiguously, but we give it a try anyways. A quote will be presumed
+    // to mark the end of the field iff it is followed by the field separator.
+    // Under this rule eol characters cannot appear inside the field.
+    // For example: <<...,"hello "world"",...>>
+    QUOTE_RULE_HYBRID,
+
+    // Fields are not quoted at all.Any quote characters appearing anywhere
+    // inside the field will be treated as any other regular characters.
+    // Example: <<...,hello "world",...>>
+    QUOTE_RULE_IGNORE_QUOTES,
+
+    QUOTE_RULE_COUNT
+};
+
+static enum quoteRule quoteRule;
```

Comment:
This is a bit confusing, should it be `typedef` instead, or perhaps the `enum` should have a slightly different name like `enum quoteRuleEnum { ... }; static enum quoteRuleEnum quoteRule;`?